src/BugsnagUnity/Client.cs

```diff
@@ -95,9 +95,27 @@ public Client(INativeClient nativeClient)
             {
                 // Async behavior is not available in a test environment
             }
+            if (!Configuration.Endpoints.IsValid)
+            {
+                CheckForMisconfiguredEndpointsWarning();
+            }
             AddBugsnagLoadedBreadcrumb();
         }
 
+        private void CheckForMisconfiguredEndpointsWarning()
+        {
+            var endpoints = Configuration.Endpoints;
+            if (endpoints.NotifyIsCustom && !endpoints.SessionIsCustom)
+            {
+                UnityEngine.Debug.LogWarning("Invalid configuration. endpoints.Notify cannot be set without also setting endpoints.Session. Events will not be sent to Bugsnag.");
+            }
+            if (!endpoints.NotifyIsCustom && endpoints.SessionIsCustom)
+            {
+                UnityEngine.Debug.LogWarning("Invalid configuration. endpoints.Session cannot be set without also setting endpoints.Notify. Sessions will not be sent to Bugsnag.");
+            }
```

Comment:
Not sure about this logic. The spec says "If `endpoints.notify` is provided by the developer, the notifier shall validate that `endpoints.session` is also provided" - but it's ambiguous whether that "provided" means "is different from the default"... @kattrali ?

Reply:
The spec states that this is to avoid "leaking data to bugsnag", I checked with Delisa and she said the idea is that we must never allow one type of endpoint to be default if the other is not. So i think this logic is correct.